### PR TITLE
[CP] Add AC show, registration, list register, fix edit form

### PR DESCRIPTION
**Pull request by NyKo24**

### CODE REVIEW

File: src/Controller/EnMarche/CitizenActionController.php [hunk 1]
```diff
@@ -36,12 +111,29 @@ public function showAction(CitizenAction $action): Response
      */
     public function exportIcalAction(CitizenAction $citizenAction): Response
     {
-        $disposition = sprintf('%s; filename=%s.ics', ResponseHeaderBag::DISPOSITION_ATTACHMENT, $citizenAction->getSlug());
+        $disposition = ResponseHeaderBag::DISPOSITION_ATTACHMENT.'; filename='.$citizenAction->getSlug().'.ics';
```
Comment (Remg) [inline src/Controller/EnMarche/CitizenActionController.php]: :trollface:

File: templates/citizen_action/_sidebar.html.twig [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<aside class="citizen_action__aside l__wrapper--aside">
+
+    <div class="citizen_action__card">
+        <h3>Quand</h3>
+        <p>{{citizen_action.beginAt|localizeddate('full', 'none')|capitalize}}</p>
+        <p>De {{citizen_action.beginAt|date('G\\hi')}} à {{citizen_action.finishAt|date('G\\hi')}}</p>
```
Comment (Remg) [inline templates/citizen_action/_sidebar.html.twig]: missing spaces

File: templates/citizen_action_manager/edit.html.twig [hunk 3]
```diff
@@ -3,3 +3,9 @@
 {% block page_title 'Éditer mon action citoyenne' %}
 
 {% block content_header 'Editer l\'action citoyenne' %}
+
+{% block citizen_action_submit_button %}
+    <div class="form__row text--center b__nudge--top-40 b__nudge--bottom-huge">
+        <button type="submit" class="btn btn--blue btn--medium">Je modifie mon action citoyenne</button>
+    </div>
+{% endblock %}
```
Comment (Remg) [inline templates/citizen_action_manager/edit.html.twig]: missing blank line

File: templates/citizen_action_manager/create.html.twig [hunk 4]
```diff
@@ -3,3 +3,9 @@
 {% block page_title 'Créer mon action citoyenne' %}
 
 {% block content_header 'Nouvelle action citoyenne' %}
+
+{% block citizen_action_submit_button %}
+    <div class="form__row text--center b__nudge--top-40 b__nudge--bottom-huge">
+        <button type="submit" class="btn btn--blue btn--medium">Je crée mon action citoyenne</button>
+    </div>
+{% endblock %}
```
Comment (Remg) [inline templates/citizen_action_manager/create.html.twig]: missing blank line

File: templates/citizen_action/_layout.html.twig [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                    </div>
+
+                    <h2 class="text--small">
+                        {% set isFull = true %}
+                        {% if citizen_action.finished %}
+                            <div class="text--body text--white">
+                                Cette action citoyenne est terminée
+                            </div>
+                        {% elseif citizen_action.cancelled %}
+                            <button class="btn btn--no-border{{ isFull?' btn--full':''}} btn--text-blue btn--disabled" disabled="disabled">
```
Comment (Remg) [inline templates/citizen_action/_layout.html.twig]: missing space `{{ isFull?' btn--full':'' }}`

File: templates/citizen_action/_layout.html.twig [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                            </div>
+                        {% else %}
+                            <div class="citizen-action-organizer text--body text--white">
+                                Organisé par un ancien adhérent
+                            </div>
+                        {% endif %}
+                    </div>
+
+                    <h2 class="text--small">
+                        {% set isFull = true %}
```
Comment (Remg) [inline templates/citizen_action/_layout.html.twig]: :thinking:

File: src/Controller/EnMarche/CitizenActionController.php [hunk 7]
```diff
@@ -2,31 +2,106 @@
...
+            throw $this->createNotFoundException(sprintf('Event "%s" is cancelled and does not accept registrations anymore', $citizenAction->getUuid()));
+        }
+
+        $command = new EventRegistrationCommand($citizenAction, $this->getUser());
+        $form = $this->createForm(EventRegistrationType::class, $command);
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $this->get(CitizenActionRegistrationCommandHandler::class)->handle($command);
+            $this->addFlash('info', $this->get('translator')->trans('citizen_initiative.registration.success'));
```
Comment (Remg) [inline src/Controller/EnMarche/CitizenActionController.php]: i guess it shouldn't be a citizen_`initiative` anymore

File: app/config/services/platform.xml [hunk 8]
```diff
@@ -371,12 +371,20 @@
...
 
         <!-- CitizenActions -->
-        <service id="AppBundle\CitizenAction\CitizenActionManager">
-            <argument type="service" id="app.repository.citizen_action"/>
-        </service>
+        <service autowire="true" id="AppBundle\CitizenAction\CitizenActionManager"/>
 
         <service autowire="true" id="AppBundle\CitizenAction\CitizenActionCommandHandler" />
 
+        <service autoconfigure="true" id="AppBundle\CitizenAction\Voter\OrganizerCitizenActionVoter" />
```
Comment (thomasbisignani) [inline app/config/services/platform.xml]: This service is already declared with :
 `<prototype namespace="AppBundle\CitizenAction\Voter\" resource="../../../src/CitizenAction/Voter/" />` in `services.xml`

File: src/Controller/EnMarche/CitizenActionController.php (hunk 7 above)
Comment (NyKo24) [reply src/Controller/EnMarche/CitizenActionController.php]: `citizen_initiative.registration.success: Votre inscription est confirmée.` is universel

File: src/Repository/CitizenActionRepository.php [hunk 9]
```diff
@@ -36,4 +36,21 @@ protected function createSlugQueryBuilder(string $slug): QueryBuilder
             ->setParameter('published', true)
         ;
     }
+
+    public function findOneCitizenActionBySlug(string $slug): ?CitizenAction
+    {
+        $query = $this
+            ->createQueryBuilder('e')
+            ->select('e', 'a', 'o')
```
Comment (thomasbisignani) [inline src/Repository/CitizenActionRepository.php]: Can you use more explicit aliases ?

Comment (NyKo24) [reply src/Repository/CitizenActionRepository.php]: No

File: templates/citizen_action/_layout.html.twig [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+                    <h1 class="citizen-action-name text--large text--white b__nudge--bottom-small text--breakword">
+                        {{- citizen_action.name -}}
+                    </h1>
+                    <div class="l__row">
+                        {% if citizen_action.organizer %}
+                            <div class="citizen-action-organizer text--body text--white">
+                                Organisé par <span class="text--bold">{{ is_granted('IS_AUTHENTICATED_FULLY') ? citizen_action.organizer.fullName : citizen_action.organizer.partialName }}</span>
+                                {% if citizen_action.citizenProject is defined %}
+                                    de l'initiative citoyenne
+                                    <a class="link--white" href="{{ path('app_citizen_project_show', {'slug': citizen_action.citizenProject.slug}) }}">
```
Comment (thomasbisignani) [inline templates/citizen_action/_layout.html.twig]: Simples quotes is useless in keys

**Pull request merged**